src/plugins/workspace/server/routes/index.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { schema } from '@osd/config-schema';
+import { CoreSetup, Logger } from '../../../../core/server';
+import { IWorkspaceClientImpl } from '../types';
+
+const WORKSPACES_API_BASE_URL = '/api/workspaces';
+
+const workspaceAttributesSchema = schema.object({
+  description: schema.maybe(schema.string()),
+  name: schema.string(),
+  features: schema.maybe(schema.arrayOf(schema.string())),
+  color: schema.maybe(schema.string()),
+  icon: schema.maybe(schema.string()),
+  defaultVISTheme: schema.maybe(schema.string()),
+});
+
+export function registerRoutes({
+  client,
+  logger,
+  http,
+}: {
+  client: IWorkspaceClientImpl;
+  logger: Logger;
+  http: CoreSetup['http'];
+}) {
+  const router = http.createRouter();
+  router.post(
+    {
+      path: `${WORKSPACES_API_BASE_URL}/_list`,
```

Comment:
The reasons is that:
1. `Workspace` is a new concept and its metadata can be stored in any other storage, for now we store that as a type of saved objects but we should provide a layer that hides the implementation detail so that we can adjust with few changes.
2. There will be some extra logic to handle workspace object in the following PRs, like ACL format, validating duplication on workspace name or creating reserved workspace, it is better that we wrap those logic inside server side and make browser call as simple as possible.

Reply:
my concern is that, even we have this client to deal with workspace related logic, since we model workspace as a type of saved object, users will still be able to use saved object APIs to access workspaces, while on those code path workspace operations doesn't have such logic in the workspace client. Maybe that can lead to un expected behaviors of workspace?

Reply:
Yeah, there are two ways that user can modify workspace metadata by bypassing workspace API:
1. Saved objects API provided by OSD: We add an ACL wrapper on saved objects client, so if user is trying to call the saved objects API, it is still under our ACL permission control.
2. update API provided by OpenSearch: Overall, we can not stop user from doing this. But in the future, few users will have permission(security plugin provides) to update `.kibana` index.